compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1389,11 +1389,16 @@ getPackVectorTileSizes(mlir::FunctionOpInterface entryPointFn,
   SmallVector<int64_t> tileSizes(op.getSourceRank(), 1);
   auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(entryPointFn);
   int64_t vectorSize = getVectorSize(entryPointFn, op.getSourceType());
+  if (!(hasAVX512fFeature(targetAttr) && isPackMatmulLHS(op))) {
+    return tileSizes;
+  }
   // TODO(#15421): Improve tile sizes selection for non f32 cases.
-  if (op.getSourceType().getElementType().isF32() &&
-      hasAVX512fFeature(targetAttr) && isPackMatmulLHS(op)) {
+  if (op.getSourceType().getElementType().isF32()) {
     tileSizes.back() = vectorSize;
   }
+  if (op.getSourceType().getElementType().isF16()) {
+    tileSizes.back() = vectorSize / 2;
+  }
```

Comment:
We should add some comment for this special logic. I think the intention is using the vector lowering strategy as same as f32 cases. To achieve it, we adjust the 2x larger vector size to half, so it has the vector shapes as same as f32 cases.